app/scripts/modules/core/src/widgets/ScopeClusterSelector.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+
+export interface IScopeClusterSelectorProps {
+  clusters: string[];
+  required?: boolean;
+  onChange: ({ clusterName }: { clusterName: string }) => void;
+  model: string;
+}
+
+export interface IScopeClusterSelectorState {
+  freeFormClusterField: boolean;
+}
+
+export class ScopeClusterSelector extends React.Component<IScopeClusterSelectorProps, IScopeClusterSelectorState> {
+  constructor(props: IScopeClusterSelectorProps) {
+    super(props);
+    const { clusters, model } = this.props;
+    const selectedNotInClusterList = !(
+      Array.isArray(clusters) &&
+      clusters.length &&
```

Comment:
Not sure you need this line since I think `some` will just return false if called on an empty array

Reply:
Ah yep good catch